components/match2/wikis/valorant/match_group_input_custom.lua

```diff
@@ -212,30 +212,39 @@ function MapFunctions.getExtraData(map, participants)
 end
 
 ---@param map table
----@param opponentCount integer
+---@param opponents table[]
 ---@return table
-function MapFunctions.getParticipants(map, opponentCount)
-	local participants = {}
+function MapFunctions.getParticipants(map, opponents)
+	local allParticipants = {}
 	local getCharacterName = FnUtil.curry(MatchGroupInputUtil.getCharacterName, AgentNames)
 
-	for opponentIdx = 1, opponentCount do
-		for _, stats, playerIdx in Table.iter.pairsByPrefix(map, 't' .. opponentIdx .. 'p', {requireIndex = true}) do
-			stats = Json.parseIfString(stats)
-
-			local participant = {
+	Array.forEach(opponents, function(opponent, opponentIndex)
+		local players = Array.mapIndexes(function(playerIndex)
+			return opponent.match2players[playerIndex] or
+				(map['t' .. opponentIndex .. 'p' .. playerIndex] and {}) or
+				nil
+		end)
+		local participants, unattachedParticipants = MatchGroupInputUtil.parseParticipants(players, function(playerIndex)
+			local stats = Json.parseIfString(map['t' .. opponentIndex .. 'p' .. playerIndex])
+			if not stats then
+				return
+			end
+			return {
 				kills = stats.kills,
 				deaths = stats.deaths,
 				assists = stats.assists,
 				acs = stats.acs,
 				player = stats.player,
 				agent = getCharacterName(stats.agent),
 			}
+		end)
+		Array.forEach(unattachedParticipants, function()
+			table.insert(participants, table.remove(unattachedParticipants, 1))
+		end)
```

Comment:
wouldn't Array.extendWith work here?

Reply:
I don't think so, as `participants` is not an 100% to be an array at this point

Reply:
okay